### PR TITLE
Feature/1173 add run search by spark app

### PR DESCRIPTION
**Pull request by DzMakatun**
#1173 Add Run search by Spark app_id

It is needed to find corresponding Run documents for given Spark History data

Add API api/runs/bySparkAppId/{spark_app_id}
which will return Run document where std_application_id OR conform_application_id equals {spark_app_id}

### CODE REVIEW

**Comment by Zejnilovic** [comment]
Just a quick question. App id is "usable" only using cluster mode as per [this code](https://github.com/apache/spark/blob/master/core/src/main/scala/org/apache/spark/SparkContext.scala#L329) (really hope I am not reading this wrongly). Aren't old runs done using client mode? Might there be a problem or duplicity?

**Comment by DzMakatun** [comment]
> Just a quick question. App id is "usable" only using cluster mode as per [this code of 2.4.4](https://github.com/apache/spark/blob/v2.4.4/core/src/main/scala/org/apache/spark/SparkContext.scala#L307) (really hope I am not reading this wrongly). Aren't old runs done using client mode? Might there be a problem or duplicity?

Good question, thnx. Here are my thoughts:
1. Spark app_id is assigned by scheduler in both cluster and client modes
2. The app_id format depends on the scheduler used (local, YARN or MESOS)
3. It is a responsibility of a scheduler to ensure the app_id is unique for the particular scheduler instance. 

To stay on the safe side, I change the implementation, so that it will return a list of all Runs with the given app_id.

File: menas/src/main/scala/za/co/absa/enceladus/menas/repositories/RunMongoRepository.scala [hunk 1]
```diff
@@ -281,6 +281,16 @@ class RunMongoRepository @Autowired()(mongoDb: MongoDatabase)
       .toFuture()
   }
 
+  def getRunBySparkAppId(appId: String): Future[Seq[Run]] = {
```
Comment (GeorgiChochov) [inline menas/src/main/scala/za/co/absa/enceladus/menas/repositories/RunMongoRepository.scala]: Could you write some tests for this in the `RunRepositoryIntegrationSuite`?

File: menas/src/main/scala/za/co/absa/enceladus/menas/controllers/RunController.scala [hunk 2]
```diff
@@ -72,6 +72,12 @@ class RunController @Autowired()(runService: RunService) extends BaseController
     runService.getRunSummariesPerDatasetVersion(datasetName)
   }
 
+  @GetMapping(Array("/bySparkAppId/{appId}"))
+  @ResponseStatus(HttpStatus.OK)
+  def getRunBySparkAppId(@PathVariable appId: String): CompletableFuture[String] = {
```
Comment (GeorgiChochov) [inline menas/src/main/scala/za/co/absa/enceladus/menas/controllers/RunController.scala]: Could you write some tests for this in the `RunApiIntegrationSuite`?

File: menas/src/main/scala/za/co/absa/enceladus/menas/repositories/RunMongoRepository.scala (hunk 1 above)
Comment (DzMakatun) [reply menas/src/main/scala/za/co/absa/enceladus/menas/repositories/RunMongoRepository.scala]: Added in 013c508 , thnx

File: menas/src/main/scala/za/co/absa/enceladus/menas/controllers/RunController.scala (hunk 2 above)
Comment (DzMakatun) [reply menas/src/main/scala/za/co/absa/enceladus/menas/controllers/RunController.scala]: Added in 1119d36, thnx

**Comment by DzMakatun** [comment]
@GeorgiChochov @Zejnilovic 
Now the SonarCloud complains about duplication because of the code which existed in the integration tests before the commit. What can be done about it?

**Comment by GeorgiChochov** [comment]
> @GeorgiChochov @Zejnilovic
> Now the SonarCloud complains about duplication because of the code which existed in the integration tests before the commit. What can be done about it?

It's fine, don't worry about sonarcloud. Maybe we'll revisit and refactor in the future.

**Pull request merged**